### PR TITLE
Update databricks diagnostic setting policy

### LINKED ISSUE #1123: Azure Databricks - Policy to deploy diagnostic settings for Databricks to Log analytics workspace - missing multiple log categories
## Versions

**Module**: AZL v2.4.1
terraform-azurerm-caf-enterprise-scale/blob/main/modules/archetypes/lib/policy_definitions/policy_definition_es_deploy_diagnostics_databricks.json

## Description
There are multiple missing log categories for Azure Databrick diagnostic settings policy. The missing categories are:
- Databricks SQL Analytics
- Databricks Genie
- Databricks Global init Scripts
- Databricks IAM Role
- Databricks MLFlow Experiment
- Databricks Feature Store
- Databricks Remote History Service
- Databricks MLFlow Acled Artifact
- Databricks DatabricksSQL
- Databricks Delta Pipelines
- Databricks Model Registry
- Databricks Repos
- Databricks Unity Catalog
- Databricks Git Credentials
- Databricks Web Terminal

The existenceCondition checks that all logs are enabled, which they are not and therefore can never be compliant.

**Steps to reproduce**

1. Assign policy to any scope
2. Deploy a Databricks resource under that scope
3. Check policy compliance
4. Check actual diagnostic setting deployed on the Databricks resource
5. 10 log categories will be enabled, and 15 categories are not enabled

**Screenshots**
Screenshot 2022-11-17 at 11 28 00
Screenshot 2022-11-17 at 11 26 04
Screenshot 2022-11-17 at 11 23 58
Screenshot 2022-11-17 at 11 24 21
Screenshot 2022-11-17 at 11 24 36

### PR DESCRIPTION
**Pull request by chrholt**
## Overview/Summary

Added missing log categories for "**Deploy Diagnostic Settings for Databricks to Log Analytics workspace**" policy definition.

## This PR fixes/adds/changes/removes

1. Fixes #1123  by adding log categories for policy definition
  - Databricks SQL Analytics
  - Databricks Genie
  - Databricks Global init Scripts
  - Databricks IAM Role
  - Databricks MLFlow Experiment
  - Databricks Feature Store
  - Databricks Remote History Service
  - Databricks MLFlow Acled Artifact
  - Databricks DatabricksSQL
  - Databricks Delta Pipelines
  - Databricks Model Registry
  - Databricks Repos
  - Databricks Unity Catalog
  - Databricks Git Credentials
  - Databricks Web Terminal 

### Breaking Changes

1. *Replace me*
3. *Replace me*

## Testing Evidence

Please provide any testing evidence to show that your Pull Request works/fixes as described and planned (include screenshots, if appropriate).

### Testing URLs

The below URLs can be updated where the placeholders are, look for `{YOUR GITHUB BRANCH NAME HERE - Remove Curly Brackets Also}` & `{YOUR GITHUB BRANCH NAME HERE - Remove Curly Brackets Also}`, to allow you to test your portal deployment experience.

> Please also replace the curly brackets on the placeholders `{}`

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fchrholt%2FEnterprise-Scale%2Fupdate-databricks-diagnostic-setting%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fchrholt%2FEnterprise-Scale%2Fupdate-databricks-diagnostic-setting%2FeslzArm%2Feslz-portal.json)

#### Azure US Gov (Fairfax)
[Deploy To Azure](https://portal.azure.us/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fchrholt%2FEnterprise-Scale%2Fupdate-databricks-diagnostic-setting%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2Fchrholt%2FEnterprise-Scale%2Fupdate-databricks-diagnostic-setting%2FeslzArm%2Ffairfaxeslz-portal.json)

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [ ] Performed testing and provided evidence.
- [ ] Updated relevant and associated documentation.
- [ ] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Comment by chrholt** [comment]
#1123

**Comment by chrholt** [comment]
@microsoft-github-policy-service agree

File: docs/wiki/Whats-new.md [hunk 1]
```diff
@@ -3,22 +3,88 @@
 - [In this Section](#in-this-section)
 - [Updates](#updates)
   - [November 2022](#november-2022)
+    - [Docs](#docs)
+    - [Tooling](#tooling)
+  - [Policy](#policy)
+  - [Other](#other)
```
Comment (jtracey93) [inline docs/wiki/Whats-new.md:9]: Could you revert all of these changes back. Suspect it's an extension that's done this 👎

We just want the month headings in the TOC

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Thanks @chrholt for the quick turnaround here.

I have left one comment and could I also ask you increment the version in `src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Diagnostics-Databricks.json` (in the metadata) to `1.2.0` as per https://github.com/Azure/azure-policy/blob/master/built-in-policies/README.md#versioning

**Comment by jtracey93** [comment]
@chrholt is this good to review again?

**Comment by chrholt** [comment]
> @chrholt is this good to review again?

Yes, I believe I figured it out now. Got a lot of unintensional changes in markdown files caused by extensions..

**Review by jtracey93 — APPROVED** [review]
LGTM

**Pull request merged**